crates/core/app/src/action_handler/transaction.rs

```diff
@@ -30,11 +30,14 @@ impl ActionHandler for Transaction {
 
         // TODO: unify code organization
         valid_binding_signature(self)?;
-        no_duplicate_spends(self)?;
-        no_duplicate_votes(self)?;
         num_clues_equal_to_num_outputs(self)?;
         check_memo_exists_if_outputs_absent_if_not(self)?;
 
+        // These should be redundant given the choice to move these checks into check_and_execute
+        // but are left here for reference and as a defense in depth.
+        no_duplicate_spends(self)?;
+        no_duplicate_votes(self)?;
```

Comment:
Slightly ambivalent on this point -- should we just delete?

Reply:
point in favor of deleting: people (me) might be tempted to add other such checks here

Reply:
sounds good, looks like our test suite is checking the exact error message, so we'll need to patch up the tests

Reply:
We probably don't want to just change the error message, but instead modify those tests to actually have the right FMD clues, to actually exercise duplicate nullifiers.